File: frontend/src/models/enums.ts
export const ConnectorStatus = {
  CREATED: 'created',
  IDLE: 'idle',
  QUEUED: 'queued',
  SYNCING: 'syncing',
  ERROR: 'error',
  PAUSED: 'paused',
  // Legacy
  STARTING: 'starting',
  VECTORIZING: 'vectorizing',
} as const;

export const ConnectorType = {
  LOCAL_FILE: 'local_file',
  LOCAL_FOLDER: 'local_folder',
  WEB: 'web',
  CONFLUENCE: 'confluence',
  SHAREPOINT: 'sharepoint',
  SQL: 'sql',
} as const;

export type ConnectorType = (typeof ConnectorType)[keyof typeof ConnectorType];

export const ScheduleType = {
  MANUAL: 'manual',
  CRON: 'cron',
} as const;

export type ScheduleType = (typeof ScheduleType)[keyof typeof ScheduleType];

export type ConnectorStatus = (typeof ConnectorStatus)[keyof typeof ConnectorStatus];

export const DocStatus = {
  PENDING: 'pending',
  IDLE: 'idle',
  PROCESSING: 'processing',
  INDEXED: 'indexed',
  FAILED: 'failed',
  SKIPPED: 'skipped',
  UNSUPPORTED: 'unsupported',
  PAUSED: 'paused',
} as const;

export type DocStatus = (typeof DocStatus)[keyof typeof DocStatus];

File: backend/app/models/enums.py
"""
Shared Enums for the application.
"""

from enum import StrEnum


class ConnectorStatus(StrEnum):
    CREATED = "created"
    IDLE = "idle"
    QUEUED = "queued"
    SYNCING = "syncing"
    ERROR = "error"
    PAUSED = "paused"
    # Legacy statuses
    STARTING = "starting"
    VECTORIZING = "vectorizing"


class DocStatus(StrEnum):
    IDLE = "idle"
    PENDING = "pending"
    PROCESSING = "processing"
    INDEXING = "indexing"
    INDEXED = "indexed"
    FAILED = "failed"
    SKIPPED = "skipped"
    PAUSED = "paused"
    UNSUPPORTED = "unsupported"


class NotificationType(StrEnum):
    """Types of notifications."""

    INFO = "info"
    SUCCESS = "success"
    WARNING = "warning"
    ERROR = "error"
    SYSTEM = "system"


class UserRole(StrEnum):
    """User Roles."""

    ADMIN = "admin"
    USER = "user"


class ConnectorType(StrEnum):
    LOCAL_FILE = "local_file"
    LOCAL_FOLDER = "local_folder"
    SQL = "sql"
    VANNA_SQL = "vanna_sql"


class ScheduleType(StrEnum):
    MANUAL = "manual"
    CRON = "cron"


class SettingGroup(StrEnum):
    """Allowed setting groups."""

    GENERAL = "general"
    AI = "ai"
    SYSTEM = "system"
    AUTH = "auth"
    STORAGE = "storage"
    NOTIFICATION = "notification"
